### PR TITLE
feat: add WithKeepSeparator option for RecursiveCharacter

### PR DESCRIPTION
**Pull request by zhangi**
Fix issue #720 
Reference Implementation In Python:  [RecursiveCharacterTextSplitter](https://api.python.langchain.com/en/latest/_modules/langchain_text_splitters/character.html#RecursiveCharacterTextSplitter)

### CODE REVIEW

File: textsplitter/options.go [hunk 1]
```diff
@@ -118,3 +120,10 @@ func WithReferenceLinks(referenceLinks bool) Option {
 		o.ReferenceLinks = referenceLinks
 	}
 }
+
+// WithKeepSeparator set the keep_separator for a text splitter.
```
[the remark this answers is not in the document]
Comment (zhangi) [reply textsplitter/options.go]: updated comments for WithKeepSeparator option.

**Review by devalexandre — CHANGES_REQUESTED** [review]
don't need update go.mod

File: textsplitter/options.go [hunk 2]
```diff
@@ -118,3 +120,14 @@ func WithReferenceLinks(referenceLinks bool) Option {
 		o.ReferenceLinks = referenceLinks
 	}
 }
+
+// WithKeepSeparator sets whether the separators should be kept in the resulting
+// split text or not. When it is set to True, the separators are included in the
+// resulting split text. When it is set to False, the separators are not included
```
Comment (devalexandre) [inline textsplitter/options.go:126]: add in description the default value

**Comment by zhangi** [comment]
> don't need update go.mod

can elaborate a bit more on the reason? The go.mod is automatically updated after my running 'go mod tidy'. There seems to be a direct dependency from [here](https://github.com/tmc/langchaingo/blob/main/llms/mistral/mistralmodel.go#L8)

File: textsplitter/options.go (hunk 2 above)
Comment (zhangi) [reply textsplitter/options.go:126]: updated the default value in comments.

**Review by tmc — APPROVED** [review]
LGTM

**Pull request merged**